toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1310,7 +1310,7 @@ func createUserWithPassword(installChroot *safechroot.Chroot, user configuration
 		}
 		isRoot = true
 	} else {
-		err = userutils.AddUser(user.Name, hashedPassword, user.UID, installChroot)
+		err = userutils.AddUser(user.Name, user.HomeDirectory, user.PrimaryGroup, hashedPassword, user.UID, installChroot)
```

Comment:
Thanks for splitting up this change from the other one!

I looked back and saw that `PrimaryGroup` was already being consumed on or around line 1240 of this file, with the call to `ConfigureUserGroupMembership()`. Is there a potential conflict with this logic? Or alternatively, is that preexisting logic sufficient or does it end up creating a primary group for the user that gets "leaked" if the user's `PrimaryGroup` isn't the same as `Name`?

Reply:
I investigated this a bit, and my conclusion is that no there is no conflict, and the preexisting logic is not sufficient because `ConfigureUserGroupMembership` is getting called after `createUserWithPassword` (that calls `AddUser`). Given that groups get created before users, the `AddUser` fails if we want to add a user to an already existing group, this forces us to us `-g` option with useradd tool.